### ISSUE TITLE
Invalid values during Coordinate conversion.

### ISSUE THREAD

**Issue by Varunvaruns9** [einsteinpy/einsteinpy#348]
🐞 **Problem**
When converting from Cartesian to BL, we may run across a divide by zero error as discussed in #343. The same error can also be encountered when converting from Cartesian to Spherical with all x, y, and z zero.

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

**Comment by Varunvaruns9** [comment]
Can we give some random value to theta in the Cartesian to Spherical case as the position doesn't depend on it when r = 0?

**Comment by shreyasbapat** [comment]
More info. I did not understand the previous comment

**Comment by Varunvaruns9** [comment]
```
pos_vec = Cartesian(0* u.km, 0 * u.km,  0 * u.km)
s_pos = pos_vec.to_spherical()
```
When converting origin `(0, 0, 0)` from Cartesian to Spherical coordinates, we get a divide by zero error as `r = z = 0` in the `cartesian_to_spherical_novel` function.

https://github.com/einsteinpy/einsteinpy/blob/66eeba86bef20cc779c644de100ddde140c7104b/src/einsteinpy/coordinates/utils.py#L33-L42

Maybe we can give some random values (or zero) to theta and phi when this happens?

**Comment by shreyasbapat** [comment]
Change the implementation to this solves this.

```
def cart2sph(x, y, z):
    hxy = np.hypot(x, y)
    r = np.hypot(hxy, z)
    el = np.arctan2(z, hxy)
    az = np.arctan2(y, x)
    return az, el, r
```

**Comment by shreyasbapat** [comment]
Similarly for reverse. What do you say?

**Comment by shreyasbapat** [comment]
```
def sph2cart(r, theta, phi):
    rcos_theta = r * np.cos(theta)
    x = rcos_theta * np.cos(phi)
    y = rcos_theta * np.sin(phi)
    z = r * np.sin(theta)
    return x, y, z
```

**Comment by Varunvaruns9** [comment]
Why do we need to change Sph->Cart?

Cartesian to Spherical seems good to me.

**Comment by shreyasbapat** [comment]
Yeah. Anyone 😂

**Comment by shreyasbapat** [comment]
Fixed in #351
